### PR TITLE
[Doc] Updating doc structure

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
Restructuring Doc dir structure.

Namely,
1. Brought `Ops` under `Common`.
2. Merged `volds.md` and `environ.md`
3. Removed `rexe.md` under `Common` as it didn't add much value.
4. Added updated docs index link in main `README.md`
5. Added a link in doc index to flow back to main menu.
6. Changed the naming convention of doc files to snake case.

Fixes: #968

Signed-off-by: srijan-sivakumar <ssivakum@redhat.com>

### CODE REVIEW

File: docs/Common/README.md [hunk 1]
```diff
@@ -3,3 +3,5 @@
 ## Index
```
Comment (nik-redhat) [inline docs/Common/README.md:3]: This header can be removed, as we have a `Common Index` header just above it.

File: docs/README.md [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 * [Common](./Common)
 * [Core](./Core)
-* [Ops](./Ops)
+* [Ops Index](./Common/Ops)
```
Comment (nik-redhat) [inline docs/README.md:5]: I think this can be removed, and we should only keep the items that are at the same level in this Index.

Comment (nik-redhat) [reply docs/README.md:5]: Also, a link to `code_considerations` can also be added. ( I forgot to add :p)

**Comment by Sheetalpamecha** [comment]
Apart from nikhil's comment, LGTM

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**